### DISCUSSION TITLE
Is it possible to use ELK not from elk-full.jar but using all dependencies in their own jar?

### DISCUSSION THREAD

**Discussion by PavelTurk** [plantuml/plantuml#2018] [Q&A]
I am trying to use ELK without elk-full because I use JPMS. All dependencies I took from https://github.com/plantuml/ELK-demo .

However, it doesn't work. I come to this point https://github.com/eclipse-elk/elk/blob/c213da54035d98bc68ae70f41cca32878d113799/plugins/org.eclipse.elk.core/src/org/eclipse/elk/core/data/LayoutAlgorithmResolver.java#L61

getting `UnsupportedConfigurationException` with message `Unable to load default layout algorithm org.eclipse.elk.layered for unconfigured node Root Node`. Any ideas how it can be solved?

**Comment by arnaudroques** [comment]
I can't provide a solution, but this appears to be the same issue as:  
https://github.com/plantuml/plantuml-server/issues/336  

Unfortunately, I couldn't resolve it either. It seems related to how ELK retrieves information (like configuration data) stored within the `.jar`, combined with some classpath intricacies.  

I would recommend raising this question with the ELK team for further insight. In any case, I’m very interested in their response as well.  

PS: Thank you for continuing to experiment with JPMS and PlantUML. I'm confident this approach will prove to be very useful in the long run.

**Reply by PavelTurk** [reply]
Thank you for your answer. I put a breakpoint here - https://github.com/plantuml/plantuml/blob/ddba6823985a886cb448535afabaa863c8bed58c/src/net/sourceforge/plantuml/elk/proxy/graph/ElkWithProperty.java#L30 and I didn't find that plantuml set property `CoreOptions.RESOLVED_ALGORITHM`. Maybe we should try to go to this direction?
